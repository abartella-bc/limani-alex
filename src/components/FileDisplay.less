// Copyright 2023 BlueCat Networks Inc.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.
@import '~@bluecateng/pelagos/less/spacing';
@import '~@bluecateng/pelagos/less/utils';
@import '~@bluecateng/pelagos/less/fonts';

.FileDisplay {
    display: flex;
    position: relative;
    flex-direction: row;
    align-items: center;
    overflow: hidden;
    border-bottom: 1px solid;
    border-bottom-color: var(--field);
    margin-bottom: @sp-08;

    &__fileName {
        flex: 1;
        padding: @sp-04 @sp-16;
        @font-12-400();
        line-height: @sp-24;
        color: var(--text-secondary);
    }

    &__delete {
        all: initial;
        color: var(--link-primary);
        @root-font();
        @font-12-400();
        margin-right: @sp-08;

        &:disabled {
            all: initial;
            color: var(--text-disabled);
            @root-font();
            @font-12-400();
            margin-right: @sp-08;
            cursor: not-allowed;
        }

        &:hover:enabled {
            all: initial;
            color: var(--link-primary-hover);
            @root-font();
            @font-12-400();
            margin-right: @sp-08;
        }
    }
}

.FileDisplay--withDownload {
    .FileDisplay__fileName {
        padding-left: @sp-02;
    }
}
