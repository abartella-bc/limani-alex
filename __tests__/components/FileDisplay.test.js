/*
Copyright 2023 BlueCat Networks Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

import { shallow } from 'enzyme';
import FileDisplay from '../../src/components/FileDisplay';

describe('FileDisplay', () => {
    describe('Rendering', () => {
        it('Render FileDisplay with default props', () => {
            const wrapper = shallow(<FileDisplay />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render FileDisplay with props, but without onDelete', () => {
            const wrapper = shallow(
                <FileDisplay
                    id='test-id-for-file-display'
                    className='test-class-A test-class-B'
                    fileName='test-file-name'
                    disabled={false}
                />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render FileDisplay with props, including onDelete, but no tooltip', () => {
            const wrapper = shallow(
                <FileDisplay
                    id='test-id-for-file-display'
                    className='test-class-A test-class-B'
                    fileName='test-file-name'
                    disabled={false}
                    onDelete={() => {}}
                />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render FileDisplay with props, including onDelete and tooltip', () => {
            const wrapper = shallow(
                <FileDisplay
                    id='test-id-for-file-display'
                    className='test-class-A test-class-B'
                    fileName='test-file-name'
                    disabled={true}
                    onDelete={() => {}}
                    deleteTooltipText='test-tooltip-for-delete-button'
                />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render FileDisplay with props, including onDownload and no tooltip', () => {
            const wrapper = shallow(
                <FileDisplay
                    id='test-id-for-file-display'
                    className='test-class-A test-class-B'
                    fileName='test-file-name'
                    disabled={true}
                    onDownload={() => {}}
                />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render FileDisplay with props, including onDownload and tooltip', () => {
            const wrapper = shallow(
                <FileDisplay
                    id='test-id-for-file-display'
                    className='test-class-A test-class-B'
                    fileName='test-file-name'
                    disabled={true}
                    onDownload={() => {}}
                    downloadTooltipText='test-tooltip-for-download-button'
                />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render FileDisplay with all props', () => {
            const wrapper = shallow(
                <FileDisplay
                    id='test-id-for-file-display'
                    className='test-class-A test-class-B'
                    fileName='test-file-name'
                    disabled={true}
                    onDelete={() => {}}
                    deleteTooltipText='test-tooltip-for-delete-button'
                    onDownload={() => {}}
                    downloadTooltipText='test-tooltip-for-download-button'
                />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
