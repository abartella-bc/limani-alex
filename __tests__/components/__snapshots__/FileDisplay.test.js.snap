// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDisplay Rendering Render FileDisplay with all props 1`] = `
<div
  className="FileDisplay test-class-A test-class-B FileDisplay--withDelete FileDisplay--withDownload"
>
  <IconButton
    className="FileDisplay__download"
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f019",
          "M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z",
        ],
        "iconName": "download",
        "prefix": "fas",
      }
    }
    id="test-id-for-file-display-download"
    onClick={[Function]}
    size="medium"
    tooltipPlacement="top"
    tooltipText="test-tooltip-for-download-button"
    type="ghost"
  />
  <div
    className="FileDisplay__fileName"
  >
    test-file-name
  </div>
  <IconButton
    className="FileDisplay__delete"
    disabled={true}
    icon={
      Object {
        "icon": Array [
          384,
          512,
          Array [
            128473,
            10005,
            10006,
            10060,
            215,
            "close",
            "multiply",
            "remove",
            "times",
          ],
          "f00d",
          "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
        ],
        "iconName": "xmark",
        "prefix": "fas",
      }
    }
    id="test-id-for-file-display-delete"
    onClick={[Function]}
    size="medium"
    tooltipPlacement="top"
    tooltipText="test-tooltip-for-delete-button"
    type="ghost"
  />
</div>
`;

exports[`FileDisplay Rendering Render FileDisplay with default props 1`] = `
<div
  className="FileDisplay"
>
  <div
    className="FileDisplay__fileName"
  />
</div>
`;

exports[`FileDisplay Rendering Render FileDisplay with props, but without onDelete 1`] = `
<div
  className="FileDisplay test-class-A test-class-B"
>
  <div
    className="FileDisplay__fileName"
  >
    test-file-name
  </div>
</div>
`;

exports[`FileDisplay Rendering Render FileDisplay with props, including onDelete and tooltip 1`] = `
<div
  className="FileDisplay test-class-A test-class-B FileDisplay--withDelete"
>
  <div
    className="FileDisplay__fileName"
  >
    test-file-name
  </div>
  <IconButton
    className="FileDisplay__delete"
    disabled={true}
    icon={
      Object {
        "icon": Array [
          384,
          512,
          Array [
            128473,
            10005,
            10006,
            10060,
            215,
            "close",
            "multiply",
            "remove",
            "times",
          ],
          "f00d",
          "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
        ],
        "iconName": "xmark",
        "prefix": "fas",
      }
    }
    id="test-id-for-file-display-delete"
    onClick={[Function]}
    size="medium"
    tooltipPlacement="top"
    tooltipText="test-tooltip-for-delete-button"
    type="ghost"
  />
</div>
`;

exports[`FileDisplay Rendering Render FileDisplay with props, including onDelete, but no tooltip 1`] = `
<div
  className="FileDisplay test-class-A test-class-B FileDisplay--withDelete"
>
  <div
    className="FileDisplay__fileName"
  >
    test-file-name
  </div>
  <IconButton
    className="FileDisplay__delete"
    disabled={false}
    icon={
      Object {
        "icon": Array [
          384,
          512,
          Array [
            128473,
            10005,
            10006,
            10060,
            215,
            "close",
            "multiply",
            "remove",
            "times",
          ],
          "f00d",
          "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
        ],
        "iconName": "xmark",
        "prefix": "fas",
      }
    }
    id="test-id-for-file-display-delete"
    onClick={[Function]}
    size="medium"
    tooltipPlacement="top"
    type="ghost"
  />
</div>
`;

exports[`FileDisplay Rendering Render FileDisplay with props, including onDownload and no tooltip 1`] = `
<div
  className="FileDisplay test-class-A test-class-B FileDisplay--withDownload"
>
  <IconButton
    className="FileDisplay__download"
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f019",
          "M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z",
        ],
        "iconName": "download",
        "prefix": "fas",
      }
    }
    id="test-id-for-file-display-download"
    onClick={[Function]}
    size="medium"
    tooltipPlacement="top"
    type="ghost"
  />
  <div
    className="FileDisplay__fileName"
  >
    test-file-name
  </div>
</div>
`;

exports[`FileDisplay Rendering Render FileDisplay with props, including onDownload and tooltip 1`] = `
<div
  className="FileDisplay test-class-A test-class-B FileDisplay--withDownload"
>
  <IconButton
    className="FileDisplay__download"
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f019",
          "M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z",
        ],
        "iconName": "download",
        "prefix": "fas",
      }
    }
    id="test-id-for-file-display-download"
    onClick={[Function]}
    size="medium"
    tooltipPlacement="top"
    tooltipText="test-tooltip-for-download-button"
    type="ghost"
  />
  <div
    className="FileDisplay__fileName"
  >
    test-file-name
  </div>
</div>
`;
